pts-core/commands/check_tests.php

```diff
@@ -0,0 +1,535 @@
+
+	// The file holding the final json results
+	const JSON_FILE = PTS_OPENBENCHMARKING_SCRATCH_PATH . 'check-tests-results.json';
+
+	// File contains a list of all the tests that have already been tested
+	const TESTED_FILES = PTS_OPENBENCHMARKING_SCRATCH_PATH . 'check-tests-tested.txt';
+
+	// Delimiter in the TESTED_FILES
+	const DELIMITER = ":::";
+
+	// Downloaded file from vendors
+	const DOWNLOADED_VENDOR_FILES = PTS_OPENBENCHMARKING_SCRATCH_PATH . "checkTestsDownloads" . "/";
+
+	// Number of child process to spawn
+	const CHILD_PROCS = 4;
+
+	// Keys used in Assocative array to store vendor data
+	const V_IDENTIFIER = 'identifier';			// name and version of the test profile
+	const V_URL = 'url';						// url from which test profile is downloaded
+	const V_DOWNLOAD_FILE = 'downloadFile';		// name of file downloaded. Usuall basename of url
+	const V_STATUS = 'status';					// http status code for the download
+	const V_DOWNLOAD_TIME = "downloadTime";		// complete time to connect to url and download
+	const V_DOWNLOAD_SIZE = "downloadSize";		// size of downloaded file
+	const V_DUPLICATE = "duplicate";			// indicates if we have already downloaded the file as part of an earlier test profile version
+	const V_REDIRECT = "redirectTo";			// If a url is redirected, this is the original url location. V_URL will contain the redirection 
+
+	/**
+	 * Determines if the test profile is valid. If invalid 'Invalid Arguement' Problem reported.
+	 * */
+	public static function argument_checks()
+	{
+		return array(
+			new pts_argument_check('VARIABLE_LENGTH_MAYBE', array('pts_test_profile', 'is_test_profile'), null)
+		);
+	}
+
+	/**
+	 * Extracts all available test profiles that will be checked. The download.xml file is extracted for
+	 * each test and test are performed on the filename, filesize, sha256 and md5 values to ensure it is
+	 * concurrent with the Service Provider (ie john-the-ripper).
+	 * 
+	 * @param $r 	Checks named test(s) only 
+	 */
+	public static function run($r)
+	{
+		if (!function_exists('curl_init')) {
+			echo pts_client::cli_colored_text("Test Failed. cURL must be installed to proceed." . PHP_EOL . PHP_EOL, 'red', true);
+		}
+
+		$startTime = microtime(true);
+
+		$procs = self::CHILD_PROCS;
+		echo pts_client::cli_colored_text('Running check-tests with ' . $procs . ' processes.' . PHP_EOL . PHP_EOL, 'green', true);
+
+		// Set up the file in which the results will be stored
+		if (file_exists(self::JSON_FILE)) {
+			if (!unlink(self::JSON_FILE))
+				echo pts_client::cli_colored_text("File " . self::JSON_FILE . " could not be deleted." . PHP_EOL . PHP_EOL, 'red', true);
+		}
+
+		if ($r == null) {
+			// extract the test-profiles that will be checked
+			$available_tests = pts_openbenchmarking::available_tests(false, true);
```

Comment:
I made an optimization to avoid needlessly checking older versions of test profiles in minor release streams, since often the new minor versions are just updating the download mirrors and such anyhow so there is no need to check prior minor versions since PTS will already handle moving to the latest compatible minor release in case of build failures or download troubles and since the download contents aren't changed anyhow post-versioning.

With the latest Git (https://github.com/phoronix-test-suite/phoronix-test-suite/commit/6237d8f8a1518a5d51e93dbde88316b9737734f8) can be enabled by setting: pts_openbenchmarking::available_tests(false, 2); (just changing the true to 2.)

It cuts down the number of tests currently to check from 1686 to 895.

E.g. Checking every test:
system/gimp-1.1.2
system/gimp-1.1.1
system/gimp-1.1.0
system/gimp-1.0.1
system/gimp-1.0.0

To just the latest minor version in each stream:
system/gimp-1.1.2
system/gimp-1.0.1

Reply:
Done